### PR TITLE
fix: check for invoice expiry

### LINKED ISSUE #767: Paying an Expired Invoice shows wrong error message
It just says “PaymentSendingFailed: Failed to send the given payment”

### PR DESCRIPTION
**Pull request by im-adithya**
Fixes #767 

This changes the `SendPaymentSync` method in transaction service to check for invoice expiry before proceeding further.

### CODE REVIEW

File: transactions/transactions_service.go [hunk 1]
```diff
@@ -206,6 +206,14 @@ func (svc *transactionsService) SendPaymentSync(ctx context.Context, payReq stri
 		return nil, err
 	}
 
+	if time.Now().After(time.Unix(int64(paymentRequest.CreatedAt+paymentRequest.Expiry), 0)) {
+		logger.Logger.WithFields(logrus.Fields{
+			"bolt11": payReq,
+		}).Errorf("Failed to decode bolt11 invoice: %v", err)
```
Comment (rolznz) [inline transactions/transactions_service.go:212]: can you update the error message

File: transactions/transactions_service.go [hunk 2]
```diff
@@ -206,6 +206,14 @@ func (svc *transactionsService) SendPaymentSync(ctx context.Context, payReq stri
 		return nil, err
 	}
 
+	if time.Now().After(time.Unix(int64(paymentRequest.CreatedAt+paymentRequest.Expiry), 0)) {
```
Comment (rolznz) [inline transactions/transactions_service.go:209]: what happens if there is no expiry in the invoice?

Comment (rolznz) [reply transactions/transactions_service.go:209]: You can omit it from the invoice - in the docs it then says it should default to 3600 seconds. But decodepay handles this for us, nice!

**Comment by rolznz** [comment]
@im-adithya you're still working on this right? it needs a new test?

**Comment by im-adithya** [comment]
@rolznz ready now!

**Review by rolznz — APPROVED** [review]
utACK

**Pull request merged**